web/client/utils/ogc/WMC/index.js

```diff
@@ -276,8 +276,44 @@ export const toMapConfig = (wmcString, generateLayersGroup = false) => {
             };
             const zoom = parseFloat(get(msTagExtractor(globalExtensions, 'zoom'), 'charContent'));
 
+            const catalogServices = msTagExtractor(globalExtensions, 'CatalogServices');
+            const selectedService = attrExtractor(catalogServices, 'selectedService');
+            const services = msTagsExtractor(catalogServices, 'Service')
+                .map(catalogService => pickAttributeValues(
+                    catalogService,
+                    'serviceName',
+                    'autoload',
+                    'title',
+                    'type',
+                    'url',
+                    'format',
+                    'metadataTemplate',
+                    'oldService',
+                    'showAdvancedSettings',
+                    'showTemplate',
+                    'hideThumbnail'
+                ))
+                .reduce((resultObj, pickedService) => ({
+                    ...resultObj,
+                    [pickedService.serviceName]: {
+                        autoload: parseBoolean(pickedService.autoload),
```

Comment:
The attributes depend on the service. More services will be added with different attributes in the future. To avoid to need to maintain this conversion for every attribute, I suggest to find a way to guess and save attributes types. Sometimes also JSON objects.
I suggest to insert all the additional attributes other than standard ones inside the tag, using tags with xsd convension. E.g.
```
 <attribute name="myNumber" type="xsd:number">2</attibute>
<attribute name="options" type="object"><![CDATA[{...}]]>
<attribute name="metadataTemplate" type="string"><![CDATA[<p>My Template</p>]]>
```
This should allow to automate conversion